airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -216,7 +220,7 @@ private void replicate(final Path jobRoot,
           });
 
       final CompletableFuture<?> readSrcAndWriteDstThread = CompletableFuture.runAsync(
-          readFromSrcAndWriteToDstRunnable(
+          srcHeartbeatTimeoutChaperone.runWithHeartbeatThread(readFromSrcAndWriteToDstRunnable(
```

Comment:
I am wondering if we have a way around this extra nesting. In the current logic, we're injecting the heartbeat chaperone in the middle which adds another level of complexity. I am wondering if a slightly more flat structure would improve the readability here.

Reply:
I am not sure how to change that to make it flatter.